src/geos.c

```diff
@@ -291,6 +291,72 @@ char check_to_wkt_compatible(GEOSContextHandle_t ctx, GEOSGeometry* geom) {
 
 #endif  // !GEOS_SINCE_3_9_0
 
+#if GEOS_SINCE_3_9_0
+
+/* Checks whether the geometry is a 3D empty geometry and, if so, create the WKT string
+ *
+ * GEOS 3.9.* is able to distiguish 2D and 3D simple geometries (non-collections). But the
+ * but the WKT serialization never writes a 3D empty geometry. This function fixes that.
+ * It only makes sense to use this for GEOS versions >= 3.9.
+ *
+ * Pending GEOS ticket: https://trac.osgeo.org/geos/ticket/1129
+ *
+ * If the geometry is a 3D empty, then the (char**) wkt argument is filled with the
+ * correct WKT string. Else, wkt becomes NULL and the GEOS WKT writer should be used.
+ *
+ * The return value is one of:
+ * - PGERR_SUCCESS
+ * - PGERR_GEOS_EXCEPTION
+ */
+char wkt_empty_3d_geometry(GEOSContextHandle_t ctx, GEOSGeometry* geom, char** wkt) {
+  char is_empty;
+  int geom_type;
+
+  is_empty = GEOSisEmpty_r(ctx, geom);
+  if (is_empty == 0) {
+    *wkt = NULL;
+    return PGERR_SUCCESS;
+  } else if (is_empty != 1) {
+    return PGERR_GEOS_EXCEPTION;
+  }
+  if (GEOSGeom_getCoordinateDimension_r(ctx, geom) == 2) {
```

Comment:
This appears to always return 2 for empty 3D geometry collections, which them prevents it from falling through to the condition below.
```
>>> pygeos.get_coordinate_dimension(pygeos.Geometry('GEOMETRYCOLLECTION Z EMPTY'))
2
```

Reply:
Thanks for checking into that.  Perhaps the thing to do is remove the fix here for GeometryCollections, since it won't be reached, and update the docstring to indicate it won't work for empty GeometryCollections?

Reply:
I updated the docstring of `to_wkt`.

Not sure what to do with the unused code. I know the parts currently cannot be reached, but I like to be forwards compatible if possible. What if empty collections turn out to be 3D at some point? If I would delete the lines, the `default:` would then lead to a `GEOSException`.

I think it is easier to leave the lines in and leave a comment about the situation. That is what I did now.